test/Interop/CxxToSwiftToCxx/consuming-cxx-struct-parameter-back-to-cxx-execution.cpp

```diff
@@ -160,6 +180,31 @@ int main() {
     SharedFRT *sfrtptr = UseCxx::returnSharedFRT(&sfrt);
     // CHECK-NEXT: retainShared
     // CHECK-NEXT: return shared frt x 2
+    SharedFRT *sfrtptr2 = UseCxx::returnSharedFRT2();
+    // No retain or release here.
+  }
+  {
+    SharedFRT sfrt;
+    sfrt.x = 4;
+    auto wrapper = UseCxx::ValueWrapper::init(&sfrt);
+    // consumeValueWrapper creates a defensive copy in the thunk.
+    UseCxx::consumeValueWrapper(wrapper);
+    // CHECK-NEXT: retainShared
+    // CHECK-NEXT: retainShared
+    // CHECK-NEXT: releaseShared
+    // CHECK-NEXT: return shared frt x 4
+    // CHECK-NEXT: releaseShared
+  }
+  {
+    SharedFRT sfrt;
+    sfrt.x = 4;
+    auto wrapper = UseCxx::ValueWrapper::init(&sfrt);
+    UseCxx::genericConsumingFunc(wrapper);
+    // CHECK-NEXT: retainShared
+    // CHECK-NEXT: retainShared
+    // CHECK-DAG: releaseShared
```

Comment:
These two lines are swapped on linux and windows, this is why I use `CHECK-DAG`.